src/jack.py

```diff
@@ -2356,6 +2546,140 @@ class CallbackExit(Exception):
     pass
 
 
+def _uuid_parse(uuid):
+    if isinstance(uuid, int):
+        return uuid
+    elif isinstance(uuid, str):
```

Comment:
This would not accept `unicode` strings on Python 2.

Maybe check `isinstance(uuid, (bytes, text_type))` here and only encode uuid, if `isinstance(uuid, text_type)`.